frontend/templates/troubleshooting/components/Causes.tsx

```diff
@@ -0,0 +1,267 @@
+import {
+   Box,
+   Heading,
+   HeadingProps,
+   Link,
+   Square,
+   useToken,
+} from '@chakra-ui/react';
+import { faList, faCircleNodes } from '@fortawesome/pro-solid-svg-icons';
+import { FaIcon } from '@ifixit/icons';
+import { useRef } from 'react';
+import { useScrollToActiveEffect } from '../toc';
+import {
+   TOCRecord,
+   useTOCBufferPxScrollOnClick,
+   useTOCContext,
+} from '../tocContext';
+
+function isIntroData(data: TOCEntry): data is IntroData & TOCRecord {
+   return data.type === 'Introduction';
+}
+function isCauseData(data: TOCEntry): data is SolutionData & TOCRecord {
+   return data.type === 'Solution';
+}
+
+function isConclusionData(data: TOCEntry): data is ConclusionData & TOCRecord {
+   return data.type === 'Conclusion';
+}
+
+export function Causes(
+   props: React.ComponentProps<typeof Box>,
+   hasRelatedPages: boolean
+) {
+   const intros: (IntroData & TOCRecord)[] = useTOCContext<TOCEntry>()
+      .getItems()
+      .filter(isIntroData);
+   const solutions: (SolutionData & TOCRecord)[] = useTOCContext<TOCEntry>()
+      .getItems()
+      .filter(isCauseData);
+   const conclusions: (ConclusionData & TOCRecord)[] = useTOCContext<TOCEntry>()
+      .getItems()
+      .filter(isConclusionData);
```

Comment:
Minor: The `getItems` runs a sort so just calling it once ups the performance
```suggestion
   const items = useTOCContext<TOCEntry>().getItems();

   const intros: (IntroData & TOCRecord)[] = items.filter(isIntroData);
   const solutions: (SolutionData & TOCRecord)[] = items.filter(isCauseData);
   const conclusions: (ConclusionData & TOCRecord)[] = items.filter(isConclusionData);
```